canal/canal.go

```diff
@@ -156,7 +156,9 @@ func (c *Canal) run() error {
 
 	if err := c.startSyncBinlog(); err != nil {
 		if !c.isClosed() {
-			log.Errorf("canal start sync binlog err: %v", err)
+			msg := fmt.Sprint("canal start sync binlog err: %v", err)
+			log.Error(msg)
+			panic(msg)
```

Comment:
Is there additional error handling we are able to do in this Canal type? I'm thinking of some examples:

- If the error code indicates the connection was severed, can we try to reconnect automatically?
- If the error code indicates we encountered a timeout attempting to establish the connection, can we do a backoff-retry for a few minutes until we successfully obtain a connection?
- Are there other error codes we can recover from automatically?